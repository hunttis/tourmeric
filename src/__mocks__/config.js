export const config = {
  default: {
    titleText: 'Foo',
    apiKey: 'Foo',
    authDomain: 'Foo',
    databaseURL: 'Foo',
    projectId: 'Foo',
    storageBucket: 'Foo',
    messagingSenderId: 'Foo',
    fileMetadataFactory: (uploadRes) => {
      const { metadata: { name, fullPath, downloadURLs } } = uploadRes;
      return {
        name,
        fullPath,
        downloadURL: downloadURLs,
      };
    },
  },
};

exports.get = function get() {
  return config.default;
};
