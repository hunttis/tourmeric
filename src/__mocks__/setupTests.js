import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16'; // eslint-disable-line

configure({ adapter: new Adapter() });
