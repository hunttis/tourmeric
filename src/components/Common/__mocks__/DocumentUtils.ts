
export default function scrollIntoview(_elementId: string) {
  console.log('scrolling');
}

export function setListener(_eventType: string, _callback: () => {}) {
  console.log('adding listener');
}

export function removeClassFromHtml(className: string) {
  console.log('removing ', className);
}

export function addClassToHtml(className: string) {
  console.log('adding ', className);
}