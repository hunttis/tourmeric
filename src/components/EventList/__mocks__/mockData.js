export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
