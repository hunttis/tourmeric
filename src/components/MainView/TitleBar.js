import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Translate, setActiveLanguage } from 'react-localize-redux';
import { isLoaded } from 'react-redux-firebase';
import Moment from 'react-moment';
import Highlights from '../HighLights/HighLights-container';

export default class TitleBar extends Component {

  changeLanguage(newLanguage) {
    this.props.dispatch(setActiveLanguage(newLanguage));
    Moment.globalLocale = newLanguage;
  }

  render() {
    const { settings, returnToFrontpage } = this.props;

    const highlightsActive = _.get(settings, 'features.highlights.active', false);

    const pageTitle = _.get(settings, 'pageTitle', 'Pagetitle not set');
    const pageSubtitle = _.get(settings, 'pageSubtitle', '');
    const settingsLoaded = isLoaded(settings);

    if (settingsLoaded) {
      return (
        <section className="hero titlebar animated">
          <div className="hero-body">
            <div className="columns is-marginless">
              <div className="column is-4 has-text-centered-mobile is-vcentered">
                {settings.activeLogo &&
                  <a onClick={() => returnToFrontpage()}>
                    <div id="logo">
                      <img src={settings.activeLogo} alt="" className="is-hidden-tablet" />

                      <figure className="image is-hidden-mobile">
                        <img src={settings.activeLogo} alt="" />
                      </figure>
                    </div>
                  </a>
                }
                {settings.pageTitle &&
                  <div id="titletext" className="title is-2">
                    {pageTitle}
                  </div>
                }
              </div>
              <div className="column is-8 button-container">
                {highlightsActive && <Highlights />}
              </div>

            </div>
            {pageSubtitle &&
              <div id="subtitletext" className="subtitle has-text-grey-light is-6 is-paddingless is-marginless">
                {pageSubtitle}
              </div>
            }
          </div>
        </section>
      );
    }

    return (
      <div className="title has-text-centered">
        <div className="level" />
        <button disabled className="button is-loading is-black"><Translate id="loading" />.</button>
      </div>
    );
  }
}

TitleBar.propTypes = {
  settings: PropTypes.object,
  dispatch: PropTypes.func,
  returnToFrontpage: PropTypes.func,
};
