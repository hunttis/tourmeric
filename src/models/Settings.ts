
export interface Settings {
  activeLocationImage?: string;
  activeLogo: string;
  browserTitle?: string;
  companyinfo?: string;
  dateFormat: string;
  features?: Features;
  introText?: string;
  location?: Location;
  openingHours?: OpeningHours;
  pageSubtitle: string;
  pageTitle: string;
  privacyPolicy?: string;
  showSponsors?: boolean;
  subtitleTextColor: string;
  theme: string;
  titleBarAngle: string;
  titleBarColor: string;
  titleBarColor2: string;
  titleBarPercentage: string;
  titleTextColor: string;
  footer?: {[key: string]: Footer};
  locationImage?: string;
}

interface Features {
  events: Feature;
  highlights: Feature;
  storeinfo: Feature;
}

interface Feature {
  active: boolean;
  introtext: boolean;
}

interface Location {
  address: string;
  directions: string;
  email: string;
  phone: string;
}


export interface OpeningHours {
  additionalinfo: string;
  monday: string;
  tuesday: string;
  wednesday: string;
  thursday: string;
  friday: string;
  saturday: string;
  sunday: string;
}

export interface Footer {
  link: string;
  image: string;
  text: string;
}
