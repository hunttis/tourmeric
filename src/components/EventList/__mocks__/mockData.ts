import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
