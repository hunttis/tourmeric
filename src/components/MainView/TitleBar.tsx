import React, { Component } from 'react';
import _ from 'lodash';
import { Translate, setActiveLanguage } from 'react-localize-redux';
import { isLoaded, isEmpty } from 'react-redux-firebase';
import Moment from 'react-moment';
import { Dispatch } from 'redux';
import Highlights from '~/components/HighLights/HighLights-container';
import OpeningHours from '../StoreInfo/OpeningHours-container';
import { Settings } from '~/models/Settings';

interface Props {
  settings: Settings;
  dispatch: Dispatch;
  returnToFrontpage: () => {};
}

export default class TitleBar extends Component<Props> {

  changeLanguage(newLanguage: string) {
    this.props.dispatch(setActiveLanguage(newLanguage));
    Moment.globalLocale = newLanguage;
  }

  render() {
    const { settings, returnToFrontpage } = this.props;

    const highlightsActive = _.get(settings, 'features.highlights.active', false);
    const introTextActive = _.get(settings, 'features.storeinfo.introtext', false);
    const introText = _.get(settings, 'introText', '');
    const introTextParagraphs = introText.split('\n');

    const pageTitle = _.get(settings, 'pageTitle', 'Pagetitle not set');
    const pageSubtitle = _.get(settings, 'pageSubtitle', '');
    const settingsLoaded = isLoaded(settings) && !isEmpty(settings);

    if (settingsLoaded) {
      return (
        <>

          {/* Hero bar only visible in desktop */}
          <section className="hero titlebar is-hidden-mobile">
            <div className="hero-body is-hidden-mobile">
              <div className="columns is-marginless">
                <div className="column is-2 is-vcentered">
                  {settings.activeLogo &&
                    <a onClick={() => returnToFrontpage()}>
                      <div id="logo" className="is-paddingless is-marginless">
                        <figure className="image">
                          <img src={settings.activeLogo} alt="" />
                        </figure>
                      </div>
                    </a>
                  }
                  {settings.pageTitle &&
                    <div id="titletext" className="title is-2">
                      {pageTitle}
                    </div>
                  }
                </div>
                <div className="column button-container">
                  {highlightsActive && <Highlights />}
                </div>
                {introTextActive &&
                  <div className="column introtext-container">
                    <div className="introtext-box">
                      {introTextParagraphs.map((paragraph: string, index: number) => <p key={`introtextparagraph-${index}`}>{paragraph}&nbsp;</p>)}
                    </div>
                  </div>
                }
              </div>
              {pageSubtitle &&
                <div id="subtitletext" className="subtitle has-text-grey-light is-6 is-paddingless is-marginless">
                  {pageSubtitle}
                </div>
              }
              <div className="is-hidden-mobile">
                <OpeningHours />
              </div>
            </div>
          </section>
        </>
      );
    }

    return (
      <div className="title has-text-centered">
        <div className="level" />
        <button disabled className="button is-loading is-black"><Translate id="loading" />.</button>
      </div>
    );
  }
}
