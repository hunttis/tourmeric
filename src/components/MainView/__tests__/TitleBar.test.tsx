/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import rrf from 'react-redux-firebase';
import TitleBar from '../TitleBar';
import { emptyMockSettings as settings } from '../../../components/EventList/__mocks__/mockData';
import { Settings } from '../../../models/Settings';

jest.mock('react-redux-firebase');
jest.mock('~/components/HighLights/HighLights-container', () => { const HighLights = () => <div />; return HighLights; });
jest.mock('~/components/StoreInfo/OpeningHours-container', () => { const OpeningHours = () => <div />; return OpeningHours; });

describe('TitleBar tests', () => {
  beforeEach(() => {
    rrf.isLoaded = jest.fn(() => true);
  });

  it('Renders titlebar before anything is loaded', () => {
    rrf.isLoaded = jest.fn(() => false);

    const testSettings: Settings = {
      // features: {
      //   highlights: {
      //     active: false,
      //     introtext: false,
      //   },
      //   events: {
      //     active: false,
      //     introtext: false,
      //   },
      //   storeinfo: {
      //     active: false,
      //     introtext: false,
      //   },
      // },
      ...settings,
    };

    const result = shallow(
      <TitleBar
        settings={testSettings}
        dispatch={jest.fn}
        returnToFrontpage={jest.fn}
      />,
    );

    expect(result.find('.is-loading').length).toBe(1);
    expect(result.find('#loading').length).toBe(1);
    expect(result).toMatchSnapshot();
  });

  it('something something', () => {
    const testSettings = {
      ...settings,
    };
    const result = shallow(<TitleBar
      settings={testSettings}
      dispatch={jest.fn}
      returnToFrontpage={jest.fn}
    />);

    expect(result).toMatchSnapshot();
  });
});
