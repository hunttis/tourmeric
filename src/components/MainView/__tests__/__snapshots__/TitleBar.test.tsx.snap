// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar tests Renders titlebar before anything is loaded 1`] = `
<div
  className="title has-text-centered"
>
  <div
    className="level"
  />
  <button
    className="button is-loading is-black"
    disabled={true}
  >
    <Translate
      id="loading"
    />
    .
  </button>
</div>
`;

exports[`TitleBar tests something something 1`] = `
<Fragment>
  <section
    className="hero titlebar is-hidden-mobile"
  >
    <div
      className="hero-body is-hidden-mobile"
    >
      <div
        className="columns is-marginless"
      >
        <div
          className="column is-2 is-vcentered"
        />
        <div
          className="column button-container"
        />
      </div>
      <div
        className="is-hidden-mobile"
      >
        <OpeningHours />
      </div>
    </div>
  </section>
</Fragment>
`;
